micrometer-core/src/main/java/io/micrometer/core/instrument/binder/db/JooqExecuteListener.java

```diff
@@ -33,7 +33,7 @@
     private final Supplier<Iterable<Tag>> queryTagsSupplier;
 
     private final Object sampleLock = new Object();
-    private volatile PMap<ExecuteContext, Timer.Sample> sampleByExecuteContext = HashTreePMap.empty();
+    private final Map<ExecuteContext, Timer.Sample> sampleByExecuteContext = new LinkedHashMap<>();
```

Comment:
`HashMap` looks like enough don't see any ordered access.

Reply:
I personally use LHM for the sake of consistency in tests, but I notice that isn't used here, so sgtm